recipe/cmake-tests/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+cmake_minimum_required(VERSION 3.17)
+
+project(diana)
+
+option(CUDA_FINDER "Which method find_package will use: CUDA or CUDAToolkit" "CUDA")
+option(WITH_ENABLE_LANGUAGE "Use enable_language(CUDA)" OFF)
+message(STATUS "WITH_ENABLE_LANGUAGE=${WITH_ENABLE_LANGUAGE} CUDA_FINDER=${CUDA_FINDER}")
+
+set(VALID_CUDA_FINDERS "CUDA" "CUDAToolkit" "OFF")
+if(NOT ${CUDA_FINDER} IN_LIST VALID_CUDA_FINDERS)
+    message(FATAL_ERROR "CUDA_FINDER must be one of: ${VALID_CUDA_FINDERS}")
+endif()
+
+file(GLOB cpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cc")
+file(GLOB gpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cu")
+file(GLOB include_files "${CMAKE_SOURCE_DIR}/src/*.hpp")
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+if(WITH_ENABLE_LANGUAGE)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 14)
+    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+    if(CUDA_FINDER)
+        find_package("${CUDA_FINDER}" REQUIRED)
+    endif(CUDA_FINDER)
+    add_definitions("-DWITHCUFILES")
+    add_executable(diana ${cpu_source_files} ${gpu_source_files})
```

Comment:
Could we check `CMAKE_CUDA_COMPILER` or something similar to ensure things are actually being built with the conda `nvcc` wrapper instead of silently using the system `nvcc`?

Reply:
Checking things below.